src/libsync/discoveryphase.cpp

```diff
@@ -443,10 +443,6 @@ void DiscoverySingleDirectoryJob::directoryListingIteratedSlot(const QString &fi
         }
         if (map.contains("data-fingerprint")) {
             _dataFingerprint = map.value("data-fingerprint").toUtf8();
-            if (_dataFingerprint.isEmpty()) {
-                // Placeholder that means that the server supports the feature even if it did not set one.
```

Comment:
As indicated by that comment that sounds dangerous (and indeed it makes for a failing test) since AFAIK it's perfectly possible to have files added on the server with no fingerprint attached to them. So the server supports the feature but no fingerprint wasn't given to a particular file (I think this can be achieved by adding a file straight in the server storage and triggering a file scan, since IIRC fingerprints are computed on the client side only such a file wouldn't have a fingerprint). A faulty client or a webdav mount on most OS could also do that by uploading a file without giving it a fingerprint. So that's why there's a difference between "no fingerprint for that file" and "no fingerprint support at all on that server". This needs to be handled somehow and I guess that explains that "" vs "[empty]".

Reply:
Well, the failing test doesn't necessarily mean the new logic is incorrect, it's more of a test and new logic are now doesn't match, and the test can be adjusted. @er-vin I'd say - we shouldn't always try to adjust the logic to match a test. The opposite is also a valid option.

Reply:
> Well, the failing test doesn't necessarily mean the new logic is incorrect, it's more of a test and new logic are now doesn't match, and the test can be adjusted. @er-vin I'd say - we shouldn't always try to adjust the logic to match a test. The opposite is also a valid option.

Well I'm not denying that but I think in that case the test is correct. Otherwise I suspect this will break some server setups. Again not an expert but my understanding is the following:
 * some servers don't support fingerprinting at all (I'm not even sure of the "at all" that seems to be also possible to have fingerprinting for some of the storage but not all, again: unsure)
 * some servers support fingerprinting and when they do...
   * some files might be there without a fingerprint (main cases in my mind would be: file added to the storage + occ file scan or files uploaded directly via a webdav client typically filemanager attached drive)
   * some files might be there with a fingerprint

Corollary of the above points (assuming they're at least partly correct):
 * we need to distinguish between files for which no fingerprint is support and for which fingerprint is missing but might be there in the future
 * some files might gain or loose fingerprints depending on what client was used last to update them, which means it needs to be known it happened and be potentially treated for conflicts

That's why I'm not shocked there's "" and "[empty]" to be handled both with a slightly different semantic. My concern by basically just killing "[empty]" is that we'll miss potential conflicts which will lead to data losses.

Again: truck load of salt on all the above. I'm just explaining why I'm very hesitant to just file it into "behavior was wrong and that's why the test is failing, let's modify the test".

Reply:
Oh one more thing, because I realize I was probably not clear before: it's indeed completely possible that there is some symmetry and that the test is broken similarly to what you're changing regarding "[empty]" not being handled properly. I'm just trying to say that it's likely there's a good reason for "[empty]" to be here in the first place. Now since you apparently found a bug involving "[empty]" it's indeed suspect no test looks for it.